bftengine/src/bftengine/SimpleClientImp.cpp

```diff
@@ -32,49 +32,45 @@ class SimpleClientImp : public SimpleClient, public IReceiver {
  public:
   SimpleClientImp(
       ICommunication* communication, uint16_t clientId, uint16_t fVal, uint16_t cVal, SimpleClientParams& p);
+  ~SimpleClientImp() override;
 
-  // SimpleClient methods
+  int sendRequest(bool isReadOnly,
+                  const char* request,
+                  uint32_t lengthOfRequest,
+                  uint64_t reqSeqNum,
+                  uint64_t timeoutMilli,
+                  uint32_t lengthOfReplyBuffer,
+                  char* replyBuffer,
+                  uint32_t& actualReplyLength) override;
 
-  virtual ~SimpleClientImp() override;
-
-  virtual int sendRequest(bool isReadOnly,
-                          const char* request,
-                          uint32_t lengthOfRequest,
-                          uint64_t reqSeqNum,
-                          uint64_t timeoutMilli,
-                          uint32_t lengthOfReplyBuffer,
-                          char* replyBuffer,
-                          uint32_t& actualReplyLength) override;
-
-  virtual int sendRequestToResetSeqNum() override;
-
-  virtual int sendRequestToReadLatestSeqNum(uint64_t timeoutMilli, uint64_t& outLatestReqSeqNum) override;
+  int sendRequestToResetSeqNum() override;
+  int sendRequestToReadLatestSeqNum(uint64_t timeoutMilli, uint64_t& outLatestReqSeqNum) override;
 
   // IReceiver methods
-
-  virtual void onNewMessage(const NodeNum sourceNode, const char* const message, const size_t messageLength) override;
-
-  virtual void onConnectionStatusChanged(const NodeNum node, const ConnectionStatus newStatus) override;
+  void onNewMessage(const NodeNum sourceNode, const char* const message, const size_t messageLength) override;
+  void onConnectionStatusChanged(const NodeNum node, const ConnectionStatus newStatus) override;
 
   // used by  MsgsCertificate
   static bool equivalent(ClientReplyMsg* r1, ClientReplyMsg* r2) {
     if (r1->reqSeqNum() != r2->reqSeqNum()) return false;
-
     if (r1->currentPrimaryId() != r2->currentPrimaryId()) return false;
-
     if (r1->replyLength() != r2->replyLength()) return false;
 
     char* p1 = r1->replyBuf();
     char* p2 = r2->replyBuf();
 
-    if (memcmp(p1, p2, r1->replyLength()) != 0) return false;
-
-    return true;
+    return (memcmp(p1, p2, r1->replyLength()) == 0);
   }
 
  protected:
-  static const uint32_t maxLegalMsgSize = 64 * 1024;  // TODO(GG): ???
-  static const uint16_t timersResolutionMilli = 50;
+  void sendPendingRequest();
+  void onMessageFromReplica(MessageBase* msg);
+  void onRetransmission();
+  void reset();
+
+ protected:
+  static const uint32_t _maxLegalMsgSize = 64 * 1024;  // TODO(GG): ???
```

Comment:
The style guide mandates trailing underscores for member variables rather than leading. I'm fine leaving this for now as it unifies the style in this file. Just mentioning in case you want to clean it up in another pass.

Reply:
Yeah, I had doubts about it.